### PR TITLE
EP-2191 - Migrate-To-GA

### PR DESCRIPTION
**Pull request by TheNoodleMoose**
This pr will be adding all of the new GTM e-commerce events for the GiveSite. I plan to work through the events in batches. The documentation that I am working through can be found [here](https://docs.google.com/spreadsheets/d/14_AEOJ_CeP_c1H9nE8TbS5wXOzUryC5E2uQBiFQJuKc/edit#gid=1140205977).

[Jira](https://jira.cru.org/browse/EP-2191)

### CODE REVIEW

**Comment by TheNoodleMoose** [comment]
This first batch adds events for add-to-cart, remove-from-cart, view-cart, and view-mini-cart. Additional events that do not require additional data have also been added.

File: src/app/analytics/analytics.factory.js
Comment (TheNoodleMoose) [inline src/app/analytics/analytics.factory.js:199]: So for this, I noticed ```callType``` wasn't actually being used so I decided to use it. They want to differentiate between viewing the mini cart and then actually going to view the cart. So where I call the analytics function for the view-mini-cart event, I pass 'mini-cart' as a parameter so we can differentiate the two events. Let me know if I should update the ```callType``` parameter name to something else.

File: src/app/profile/yourGiving/editRecurringGifts/step4/confirmRecurringGifts.component.js
Comment (TheNoodleMoose) [inline src/app/profile/yourGiving/editRecurringGifts/step4/confirmRecurringGifts.component.js:35]: A lot of these events don't need any additional data, so I just used the ```trackGTM``` function and changed the event name to align with google analytics.

File: src/common/services/session/sessionModal.service.js
Comment (TheNoodleMoose) [inline src/common/services/session/sessionModal.service.js:68]: I didn't know if these needed to be two separate events or not. There wasn't anything in the documentation so I just used the same analytics call for both. I'm going to bring it up to Clark though.

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [inline src/app/analytics/analytics.factory.js:122]: Do you need to lowercase a number?

Comment (TheNoodleMoose) [reply src/app/analytics/analytics.factory.js:122]: That I do not, haha. I'll remove that.

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [inline src/app/analytics/analytics.factory.js:195]: So, `cartView` isn't currently called anywhere. Maybe if you aren't going to use the `cartData` parameter, it should be removed from the definition?

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:199]: Since the current code doesn't use it, you could make it a boolean if you wanted

Comment (TheNoodleMoose) [reply src/app/analytics/analytics.factory.js:199]: Yeah, that sounds good. I'll change it to ```isMiniCart``` and then have a default value of false.

File: src/common/components/nav/navCartIcon.component.js
Comment (wrandall22) [inline src/common/components/nav/navCartIcon.component.js:14]: You also need to add this to the `.module` piece at the bottom (see `stopStartRecurringGifts.modal.component.js`)

Comment (wrandall22) [reply src/common/components/nav/navCartIcon.component.js:14]: And you need to import it at the top I believe.

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [inline src/app/analytics/analytics.factory.js:590]: So far you aren't sending `data` in any of the calls to this function.  Are you planning to use it in the future?

File: src/common/components/nav/navCartIcon.component.js
Comment (TheNoodleMoose) [reply src/common/components/nav/navCartIcon.component.js:14]: Okay went ahead and added that.

File: src/app/analytics/analytics.factory.js
Comment (TheNoodleMoose) [reply src/app/analytics/analytics.factory.js:590]: Yeah, good point. Any event that needs data I'll probably just make a custom analytics function for so I'll remove the data parameter.

File: src/app/analytics/analytics.factory.js
Comment (wrandall22) [inline src/app/analytics/analytics.factory.js:199]: This block doesn't really make sense.

Comment (TheNoodleMoose) [reply src/app/analytics/analytics.factory.js:199]: Yeah, I'm not sure what the logic before was, but the event being fired is an ```aa-view-minicart``` event. So I figured we would only want that to be called if ```isMiniCart``` is true?

Comment (wrandall22) [reply src/app/analytics/analytics.factory.js:199]: It was never called, so why not just remove the block?

Comment (TheNoodleMoose) [reply src/app/analytics/analytics.factory.js:199]: Ah good point, yep will do!

**Comment by wrandall22** [comment]
This is probably ok to push to staging, but you should add more tests before I approve the PR.

**Comment by TheNoodleMoose** [comment]
@wrandall22 Here's the second batch of events. I made sure to add tests for these events while I was adding them. I'm going to make another commit to add tests for the first batch.

**Comment by wrandall22** [comment]
It looks fine to push to staging, but looks like it wants more tests for prod.

**Comment by TheNoodleMoose** [comment]
Okay, I'll add more tests, I think there was a big section that didn't have tests initially for AA and that's why there's such a big miss in coverage.

**Comment by TheNoodleMoose** [comment]
Just pushed it to staging

**Review by wrandall22 — APPROVED** [review]

**Comment by TheNoodleMoose** [comment]
@wrandall22 Just got the green light from Clark that everything looks good on stage! 🥳 Should I merge it in now or wait till the morning?

**Comment by wrandall22** [comment]
I would wait at this point.

**Comment by TheNoodleMoose** [comment]
Sounds good, I'll merge in the morning.

**Pull request merged**
